### PR TITLE
delly: init at 0.8.1

### PR DESCRIPTION
**Pull request by scalavision**
###### Things done

- [ x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [ x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ x] Ensured that relevant documentation is up to date
- [ x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by scalavision** [comment]
Delly is a very much used tool used to find structural variants within dna sequenced, short read data. It would be very nice to have it in nixpkgs.

File: pkgs/applications/science/biology/delly/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchgit, htslib, zlib, bzip2, lzma, ncurses, boost }:
+
+stdenv.mkDerivation rec {
+
+  name = "${pname}-${version}";
+  pname = "delly";
+  version = "0.8.1";
+
+  src = fetchgit {
```
Comment (markuskowa) [inline pkgs/applications/science/biology/delly/default.nix]: Please use `fetchFromGitHub` here.

File: pkgs/applications/science/biology/delly/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      url = "https://github.com/dellytools/${pname}.git";
+      rev = "1da97218e5324c00acb2bb33b0e7985fad4d8079";
+      sha256 = "0ip6x1g5dr7j8w3qhvg4d1p4ix88mcx0jjb01zsnaaxq3gbh3b09";
+  };
+
+  buildInputs = [ zlib htslib bzip2 lzma ncurses boost ];
+
+  buildPhase = ''
```
Comment (markuskowa) [inline pkgs/applications/science/biology/delly/default.nix]: `preBuild` is the better choice here:
```suggestion
  preBuild = ''
     export EBROOTHTSLIB=${htslib}
  '';   
```

File: pkgs/applications/science/biology/delly/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  '';
+
+  installPhase = ''
+      mkdir -p $out/bin
+      cp src/delly $out/bin/delly    
+      export PATH=$PATH:$out/bin/delly
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Structural variant caller";
```
Comment (markuskowa) [inline pkgs/applications/science/biology/delly/default.nix]: Maybe something like that?
```suggestion
    description = "Structural variant caller for DNA sequence matching";
```
That would give it more context if viewed in a list of packages.

File: pkgs/applications/science/biology/delly/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  buildPhase = ''
+      export EBROOTHTSLIB=${htslib}
+      make
+  '';
+
+  installPhase = ''
+      mkdir -p $out/bin
+      cp src/delly $out/bin/delly    
+      export PATH=$PATH:$out/bin/delly
```
Comment (markuskowa) [inline pkgs/applications/science/biology/delly/default.nix]: What is the purpose of the `export`?

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/delly/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  buildInputs = [ zlib htslib bzip2 lzma ncurses boost ];
+
+  buildPhase = ''
+      export EBROOTHTSLIB=${htslib}
+      make
+  '';
+
+  installPhase = ''
+      mkdir -p $out/bin
+      cp src/delly $out/bin/delly
```
Comment (veprbl) [inline pkgs/applications/science/biology/delly/default.nix]: Can replace these two lines
```suggestion
      install -Dm555 src/delly $out/bin/delly    
```

File: pkgs/applications/science/biology/delly/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  buildInputs = [ zlib htslib bzip2 lzma ncurses boost ];
+
+  buildPhase = ''
+      export EBROOTHTSLIB=${htslib}
+      make
+  '';
+
+  installPhase = ''
```
Comment (veprbl) [inline pkgs/applications/science/biology/delly/default.nix]: When overriding a stdenv phase it's a good practice to call the standard hooks:
```nix
stdenv.mkDerivation {
  # ...

  installPhase = ''
    runHook preInstall

    # your installation procedure here

    runHook postInstall
  '';

  # ...
}
```

File: pkgs/applications/science/biology/delly/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/applications/science/biology/delly/default.nix]: No need to use export, just set the attribute:
```nix
stdenv.mkDerivation {
  # ...

  EBROOTHTSLIB = htslib;

  # ...
}
```

File: pkgs/applications/science/biology/delly/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchgit, htslib, zlib, bzip2, lzma, ncurses, boost }:
+
+stdenv.mkDerivation rec {
+
+  name = "${pname}-${version}";
```
Comment (veprbl) [inline pkgs/applications/science/biology/delly/default.nix]: Remove this line, this is already done inside the `mkDerivation`.

**Comment by mmahut** [comment]
After fixing the issues above, please squash your commits too.

File: pkgs/applications/science/biology/delly/default.nix (hunk 7 above)
Comment (scalavision) [reply pkgs/applications/science/biology/delly/default.nix]: I removed the `name` attribute, is this the reason why the build failed?

**Comment by scalavision** [comment]
Thanks for all the nice feedback! I've removed buildPhase, only using prePhase. Also I've cleaned up the installPhase, which seems to work.

Anyone knows why the checkMeta validation doesn't work? It says something about scylla-db. I don't think I have anything related to that.

**Comment by veprbl** [comment]
Eval should be fixed by c0d5833156cd32086642207fbb026ab772d16822

**Comment by veprbl** [comment]
@GrahamcOfBorg build delly

**Review by veprbl — APPROVED** [review]

**Pull request merged**